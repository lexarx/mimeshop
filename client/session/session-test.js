jest.mock('api-request');
import { getSessionId, startSession } from './session'
import apiRequest from 'api-request'

test('starts a new session and saves session ID', async () => {
	apiRequest.mockImplementation((method, url) => {
		if (method === 'POST' && url === 'start-session') {
			return {sessionId: '123'};
		}
	});
	await startSession();
	expect(getSessionId()).toBe('123');
	expect(localStorage.getItem('sessionId')).toBe('123');
});

test('sends saved session ID', async () => {
	localStorage.setItem('sessionId', '345');
	apiRequest.mockReturnValue({ sessionId: '345' });
	await startSession();
	expect(apiRequest).toHaveBeenCalledWith('POST', 'start-session', { sessionId: '345' });
});