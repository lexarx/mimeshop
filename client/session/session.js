import apiRequest from 'api-request';

let sessionId;

export async function startSession() {
	const response = await apiRequest('POST', 'start-session', {
		sessionId: localStorage.getItem('sessionId')
	});
	sessionId = response.sessionId;
	localStorage.setItem('sessionId', sessionId);
}

export function getSessionId() {
	return sessionId;
}