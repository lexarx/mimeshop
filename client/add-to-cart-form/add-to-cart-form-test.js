jest.mock('api-request');
jest.mock('session/session');
import React from 'react'
import AddToCartForm from './add-to-cart-form'
import { mount, render } from 'enzyme'
import apiRequest from 'api-request'
import { getSessionId } from 'session/session';

test('displays add to cart form', () => {
	const addToCartForm = render(<AddToCartForm/>);
	expect(addToCartForm).toMatchSnapshot();
});

test('sends server request', () => {
	const addToCardForm = mount(<AddToCartForm productId="123"/>);
	const quantity = addToCardForm.find('input');
	quantity.instance().value = '5';
	quantity.simulate('change');
	apiRequest.mockReturnValue({});
	getSessionId.mockReturnValue('s1');
	addToCardForm.find('button').simulate('click');
	expect(apiRequest).toHaveBeenCalledWith('POST', 'add-to-cart', {
		productId: '123',
		quantity: '5',
		sessionId: 's1'
	});
});