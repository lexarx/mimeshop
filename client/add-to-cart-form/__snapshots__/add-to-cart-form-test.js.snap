// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays add to cart form 1`] = `
Array [
  <input
    class="quantity"
    placeholder="Quantity"
    value=""
  />,
  <button
    class="addButton"
  >
    Add to cart
  </button>,
]
`;
